src/Chapter1.hs

```diff
@@ -640,8 +642,10 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Integral a=> a->a 
+firstDigit n = if n' <10 then  n' else firstDigit(div n' 10)
```

Comment:
One minor stylistic suggestion: operators and function arguments are usually separated by spaces for better visibility.
```suggestion
firstDigit n = if n' < 10 then  n' else firstDigit (div n' 10)
```